src/App.js

```diff
@@ -1,10 +1,78 @@
+import { useState, useEffect, useCallback } from "react";
+import TodoList from "./TodoList";
+import TodoInputForm from "./TodoInputForm";
+import styled from "styled-components";
+
+const Container = styled.div`
+  height: 100vh;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  font-family: "Noto Sans KR", sans-serif;
+  background: linear-gradient(to bottom right, #7f7fd5, #86a8e7, #91eae4);
+`;
+const Box = styled.main`
+  width: 350px;
+  height: 600px;
+  background-color: white;
+  border-radius: 20px;
+  box-shadow: 1px 1px 30px grey;
+`;
+
+const App = () => {
+  const [todoList, setTodoList] = useState([]);
+  const [doneList, setDoneList] = useState([]);
+
+  // save to localStorage
+  const _saveLocalStorage = useCallback((type, list) => {
+    localStorage.setItem(type, JSON.stringify(list));
+  }, []);
+
+  // 렌더링 시 localStorage에서 todo list 불러오기
+  useEffect(() => {
+    const todoStorage = localStorage.getItem("todoList");
+    const doneStorage = localStorage.getItem("doneList");
+
+    // localStorage에 저장된 list가 있는지 확인
+    if (todoStorage) {
+      const loadTodo = JSON.parse(todoStorage);
+      setTodoList(loadTodo);
+    }
+    if (doneStorage) {
+      const loadDone = JSON.parse(doneStorage);
+      setDoneList(loadDone);
+    }
+  }, []);
+
+  // list가 변할 때마다 localStorage에 list 저장
+  useEffect(() => {
+    _saveLocalStorage("todoList", todoList);
+  }, [todoList]);
+  useEffect(() => {
+    _saveLocalStorage("doneList", doneList);
+  }, [doneList]);
```

Comment:
dependency array 관련 에러가 뜨네요
참고자료에 첨부된 `useEffect 완벽 가이드` 읽어 보시면 도움 많이 되실 것 같아요!

```suggestion
  useEffect(() => {
    _saveLocalStorage("todoList", todoList);
  }, [todoList, _saveLocalStorage]);
  useEffect(() => {
    _saveLocalStorage("doneList", doneList);
  }, [doneList, _saveLocalStorage]);
```

Reply:
넵 감사합니다:smile: